src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -23,34 +22,28 @@ internal class VsProjectJsonToPackageReferenceMigrator : IVsProjectJsonToPackage
         public VsProjectJsonToPackageReferenceMigrator(Lazy<IVsSolutionManager> solutionManager)
         {
             _solutionManager = solutionManager;
-            if(solutionManager == null)
+            if (solutionManager == null)
             {
                 throw new ArgumentNullException(nameof(solutionManager));
             }
         }
-        public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackageReference(string projectUniqueName)
+        public Task<object> MigrateProjectJsonToPackageReferenceAsync(string projectUniqueName)
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-            if(string.IsNullOrEmpty(projectUniqueName))
+            if (string.IsNullOrEmpty(projectUniqueName))
             {
                 throw new ArgumentNullException(nameof(projectUniqueName));
             }
 
-            if(!File.Exists(projectUniqueName))
+            if (!File.Exists(projectUniqueName))
             {
                 throw new FileNotFoundException(nameof(projectUniqueName));
             }
 
-            return  NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => 
-            {
-                await TaskScheduler.Default;
-                return await MigrateProjectToPackageRefAsync(projectUniqueName);
-            });
-            
+            return MigrateProjectToPackageRefAsync(projectUniqueName);
         }
 
-        private async Task<IVsProjectJsonToPackageReferenceMigrateResult> MigrateProjectToPackageRefAsync(string projectUniqueName)
-        {            
+        private async Task<object> MigrateProjectToPackageRefAsync(string projectUniqueName)
+        {
             var project = _solutionManager.Value.GetDTEProject(projectUniqueName);
```

Comment:
`GetDTEProject` isn't IVs or DTE service, it's NuGet api which make sure to switch to UI thread whenever needed.